raiden/ui/startup.py

```diff
@@ -0,0 +1,361 @@
+            "Please provide them on the command line or in the configuration file.",
+            fg='red',
+        )
+        sys.exit(1)
+
+    try:
+        token_network_registry = blockchain_service.token_network_registry(
+            tokennetwork_registry_contract_address or to_canonical_address(
+                contracts[CONTRACT_TOKEN_NETWORK_REGISTRY]['address'],
+            ),
+        )
+    except ContractVersionMismatch as e:
+        handle_contract_version_mismatch(e)
+    except AddressWithoutCode:
+        handle_contract_no_code('token network registry', tokennetwork_registry_contract_address)
+    except AddressWrongContract:
+        handle_contract_wrong_address(
+            'token network registry',
+            tokennetwork_registry_contract_address,
+        )
+
+    try:
+        secret_registry = blockchain_service.secret_registry(
+            secret_registry_contract_address or to_canonical_address(
+                contracts[CONTRACT_SECRET_REGISTRY]['address'],
+            ),
+        )
+    except ContractVersionMismatch as e:
+        handle_contract_version_mismatch(e)
+    except AddressWithoutCode:
+        handle_contract_no_code('secret registry', secret_registry_contract_address)
+    except AddressWrongContract:
+        handle_contract_wrong_address('secret registry', secret_registry_contract_address)
+
+    # If services contracts are provided via the CLI use them instead
+    if user_deposit_contract_address is not None:
+        contracts[CONTRACT_USER_DEPOSIT] = user_deposit_contract_address
+    if service_registry_contract_address is not None:
+        contracts[CONTRACT_SERVICE_REGISTRY] = (
+            service_registry_contract_address
+        )
+
+    user_deposit = None
+    should_use_user_deposit = (
+        environment_type == Environment.DEVELOPMENT and
+        ID_TO_NETWORKNAME.get(node_network_id) != 'smoketest' and
+        CONTRACT_USER_DEPOSIT in contracts
+    )
+    if should_use_user_deposit:
+        try:
+            user_deposit = blockchain_service.user_deposit(
+                user_deposit_contract_address or to_canonical_address(
+                    contracts[CONTRACT_USER_DEPOSIT]['address'],
+                ),
+            )
+        except ContractVersionMismatch as e:
+            handle_contract_version_mismatch(e)
+        except AddressWithoutCode:
+            handle_contract_no_code('user deposit', user_deposit_contract_address)
+        except AddressWrongContract:
+            handle_contract_wrong_address('user_deposit', user_deposit_contract_address)
+
+    service_registry = None
+    if routing_mode == RoutingMode.PFS:
+
+        if environment_type == Environment.PRODUCTION:
+            click.secho(
+                'Requested production mode and PFS routing mode. This is not supported',
+                fg='red',
+            )
+            sys.exit(1)
+
+        if CONTRACT_SERVICE_REGISTRY not in contracts:
```

Comment:
Can you check if `--pathfinding-service-address` is given, and otherwise make it possible to skip this?

See https://github.com/raiden-network/raiden/issues/3740

Reply:
Surething. Will also add a test for this then.